### DISCUSSION TITLE
Uncaught ReferenceError: currentlyReorderingStatus is not defined

### DISCUSSION THREAD

**Discussion by jclee100** [rappasoft/laravel-livewire-tables#1432] [Q&A]
Not sure what am I missing. I can filter but I cannot sort. However, after apply filters, sort starts working.

I am using the latest version with Laravel 10, PHP 8.1+, 

`Uncaught ReferenceError: currentlyReorderingStatus is not defined`
`Expression: "currentlyReorderingStatus ? 'laravel-livewire-tables-reorderingMinimised' : ''" ....`

**Comment by lrljoe** [comment]
Typically this is one of a couple of things:
If you have previously published the package views, I'd advise removing them.  You shouldn't need to publish the views for this package, and it's not advised.

If you are bundling the assets rather than injecting them, then make sure to update your bundle

**Comment by jclee100** [comment]
I don't think I published the views and I am not bundling assets.

I see the scripts injected in head

```
<head><!-- Rappasoft Core Table Styles -->
    <link href="/rappasoft/laravel-livewire-tables/core.min.css" rel="stylesheet"> <!-- Rappasoft Table Third Party Styles -->
    <link href="/rappasoft/laravel-livewire-tables/thirdparty.css" rel="stylesheet">
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="csrf-token" content="7kOYmd7fTxDxROnub2H1TXuwnwXNODvl3jBDb62v">

        <title>Laravel</title>

        <!-- Fonts -->
        <link rel="preconnect" href="https://fonts.bunny.net">
        <link href="https://fonts.bunny.net/css?family=figtree:400,500,600&amp;display=swap" rel="stylesheet">

        <!-- Scripts -->
        <script type="module" src="http://127.0.0.1:5174/@vite/client" data-navigate-track="reload"></script><link rel="stylesheet" href="http://127.0.0.1:5174/resources/css/app.css" data-navigate-track="reload"><script type="module" src="http://127.0.0.1:5174/resources/js/app.js" data-navigate-track="reload"></script>    <!-- Rappasoft Core Table Scripts -->
<script src="/rappasoft/laravel-livewire-tables/core.min.js"></script> <!-- Rappasoft Third Party Scripts -->
<script src="/rappasoft/laravel-livewire-tables/thirdparty.min.js"></script>
<!-- Livewire Styles --><style>[wire\:loading], [wire\:loading\.delay], [wire\:loading\.inline-block], [wire\:loading\.inline], [wire\:loading\.block], [wire\:loading\.flex], [wire\:loading\.table], [wire\:loading\.grid], [wire\:loading\.inline-flex] {display: none;}[wire\:loading\.delay\.none], [wire\:loading\.delay\.shortest], [wire\:loading\.delay\.shorter], [wire\:loading\.delay\.short], [wire\:loading\.delay\.default], [wire\:loading\.delay\.long], [wire\:loading\.delay\.longer], [wire\:loading\.delay\.longest] {display:none;}[wire\:offline] {display: none;}[wire\:dirty]:not(textarea):not(input):not(select) {display: none;}:root {--livewire-progress-bar-color: #2299dd;}[x-cloak] {display: none;}</style><style>/* Make clicks pass-through */

```

I see `currentlyReorderingStatus` in the injected core.min.js but the console says `ReferenceError: currentlyReorderingStatus is not defined`. I am guessing the method is called before it's being initialised/loaded which can explain why it works after performing some manual actions such as filtering.

Still not sure what I am doing wrong.

**Reply by lrljoe** [reply]
Make sure to update to the latest version of the package.

Let me know if this is still throwing any errors for you, as there's a few different fix/approaches for similar issues.

**Reply by jclee100** [reply]
It was v3+ and I have now updated to 3.1.3. Still the same errors.

```
livewire.js?id=2b77c128:1234 Alpine Expression Error: currentlyReorderingStatus is not defined

Expression: "currentlyReorderingStatus ? 'laravel-livewire-tables-reorderingMinimised' : ''"
```

**Reply by lrljoe** [reply]
Check in resources/views/vendor to make sure the table views aren't in there.

If they're not then I'll take another closer look.

If you open the browser straight to the table, or refresh the page with the table, do you still get the error?

**Reply by jclee100** [reply]
> Check in resources/views/vendor to make sure the table views aren't in there.

Just checked. I don't have a vendors directory in views.

> If you open the browser straight to the table, or refresh the page with the table, do you still get the error?

Yes error still occurs.

**Comment by ChrisThompsonTLDR** [comment]
@lrljoe I am also encountering this issue after updating Laravel and a few packages.

I have v3.1.3 of this package installed.

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/348801/17d746ab-6b69-4f73-95d2-200cf76d41e0)

I do not have the package's view files in my application

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/348801/0f2fb777-8f84-4014-b0eb-9e53331d7802)

I have asset caching disabled in my browser

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/348801/d0a53f12-17e3-4332-9a2a-e99933cf2cc7)

I still receive the error

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/348801/814cc1d1-277a-4c7c-999e-b56f283d61a6)

**Reply by ChrisThompsonTLDR** [reply]
Digging deeper, I think this is an issue with the bundling of assets, specifically Alpine.

**Reply by janglapuk** [reply]
I think the problems are in their public assets changes for current version (v3.1.3). Make sure to check `resources` directory in this repository.

If you enabling `inject_core_assets_enabled` and `inject_third_party_assets_enabled` in `config/livewire-tables.php`, then:

1. `script_base_path` is `/rappasoft/laravel-livewire-tables` by default, should be `/rappasoft/livewire-tables` (or `/vendor/rappasoft/livewire-tables`)
2. Asset names are `core*.css`, `thirdparty*.css`, `core*.js`, and `thirdparty*.js` by default, should be `laravel-livewire-tables*.css` and `laravel-livewire-tables*.js`. Still don't know how to override this. Need bugfixes in https://github.com/rappasoft/laravel-livewire-tables/blob/c5f0539bcd3f317a0ac0b07d46695158c6c03a52/src/Mechanisms/RappasoftFrontendAssets.php#L25-L57

As a temporary solution, you need to disabling `inject_core_assets_enabled` and `inject_third_party_assets_enabled`, and includes the assets manually in your blade.

```html
<link rel="stylesheet" href="{{ asset('vendor/rappasoft/livewire-tables/css/laravel-livewire-tables.min.css') }}">
<link rel="stylesheet" href="{{ asset('vendor/rappasoft/livewire-tables/css/laravel-livewire-tables-thirdparty.min.css') }}">
...
<script src="{{ asset('vendor/rappasoft/livewire-tables/js/laravel-livewire-tables.min.js') }}"></script>
<script src="{{ asset('vendor/rappasoft/livewire-tables/js/laravel-livewire-tables-thirdparty.min.js') }}"></script>
```

**Reply by lrljoe** [reply]
Thanks for the detailed description, will get a fix out in the next day hopefully.

**Comment by StuBlackett2022** [comment]
Hi @lrljoe any updates on this. We seem to have a clash between livewire and this Vue tables package
